run_dir/static/js/projects.js

```diff
@@ -498,6 +508,49 @@ $('body').on('click hidden.bs.dropdown', '.rBtngp2', function(event){
   read_current_filtering();
 });
 
+//Saving the filtering/sorting to user-defined preset
+$(document).on('click', '#saveFilter', function() {
+  var presetFilterName = '';
+  if ($('#user_presets_dropdown .dropdown-toggle').hasClass('active')){
+    //if user-defined preset selected, save to old preset
+    presetFilterName=$('#user_presets_dropdown .dropdown-toggle').text();
+    if (sort_preset !== presetFilterName){
+      //if selected preset does not match the loaded preset, otherwise in won't apply to the correct preset
+      alert('Click "Load Table" to save to Preset');
+    }
+    else {
+      //save
+      var table_filter = $('#project_table').DataTable().state()
+      var userPage_api_url = "/api/v1/presets?savefilter="+$.trim(presetFilterName);
+      $.ajax({
+        type: 'POST',
+        dataType: 'json',
+        url: userPage_api_url,
+        data: JSON.stringify(table_filter),
+        error: function(xhr, textStatus, errorThrown) {
+          alert('There was an error in saving the preset: '+errorThrown);
+          $('#saveFilter').removeClass('disabled').text('Apply');
```

Comment:
The button text shouldn't be `Apply` if there is an error with saving the filtering.